core/src/com/fourinachamber/fortyfive/map/detailMap/SeededMapGenerator.kt

```diff
@@ -74,15 +74,17 @@ class SeededMapGenerator(
         val deadEndsWithoutEvents = nodesWithoutEvents.filter { it.edgesTo.size == 1 }.toMutableList()
         val sortedFixedEvents = restrictions.fixedEvents.sortedBy { !it.second }
         nodesWithoutEvents.removeAll(deadEndsWithoutEvents)
-        for (curEvent in sortedFixedEvents) {
+        for ((event, isDeadEnd, nodeTexture) in sortedFixedEvents) {
             if (nodesWithoutEvents.isEmpty() && deadEndsWithoutEvents.isEmpty()) break
-            if ((curEvent.second && deadEndsWithoutEvents.isNotEmpty()) || nodesWithoutEvents.isEmpty()) {
+            if ((isDeadEnd && deadEndsWithoutEvents.isNotEmpty()) || nodesWithoutEvents.isEmpty()) {
                 val curNode = deadEndsWithoutEvents.random(rnd)
-                curNode.event = curEvent.first
+                curNode.event = event
+                curNode.nodeTexture = nodeTexture
                 deadEndsWithoutEvents.remove(curNode)
             } else {
                 val curNode = nodesWithoutEvents.random(rnd)
-                curNode.event = curEvent.first
+                curNode.event = event
+                curNode.nodeTexture = nodeTexture
```

Comment:
before you merge, pull main beforehand and make sure these parts don't fail, because i think i changed something here as well, but not sure